ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -416,9 +426,19 @@ private static AddPartitionDesc getBaseAddPartitionDescFromPartition(
     partDesc.setSerdeParams(partition.getSd().getSerdeInfo().getParameters());
     partDesc.setBucketCols(partition.getSd().getBucketCols());
     partDesc.setSortCols(partition.getSd().getSortCols());
-    partDesc.setLocation(new Path(fromPath,
-        Warehouse.makePartName(tblDesc.getPartCols(), partition.getValues())).toString());
-    return partsDesc;
+    if (replicationSpec.isInReplicationScope() && tblDesc.isExternal()) {
```

Comment:
If it is migration flow (managed to external), then the location should be default external table path got from config in target warehouse. It shouldn't preserve the source path.

Reply:
yeh going to fix this, also looks like the wrong class for test TestReplicationScenariosMigration is inherited, should have been TestReplicationWithTableMigration, going to change that as well.